RetailCoder.VBE/UI/RegexAssistant/RegexAssistantViewModel.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Rubberduck.RegexAssistant;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace Rubberduck.UI.RegexAssistant
+{
+    public class RegexAssistantViewModel : ViewModelBase
+    {
+        public RegexAssistantViewModel()
+        {
+            _pattern = string.Empty;
+            RecalculateDescription();
+        }
+
+        public bool GlobalFlag {
+            get
+            {
+                return _globalFlag;
+            }
+            set
+            {
+                _globalFlag = value;
+                RecalculateDescription();
+            }
+        }
+        public bool IgnoreCaseFlag
+        {
+            get
+            {
+                return _ignoreCaseFlag;
+            }
+            set
+            {
+                _ignoreCaseFlag = value;
+                RecalculateDescription();
+            }
+        }
+        public string Pattern
+        {
+            get
+            {
+                return _pattern;
+            }
+            set
+            {
+                _pattern = value;
+                RecalculateDescription();
+            }
+        }
+
+        private string _description;
+        private bool _globalFlag;
+        private bool _ignoreCaseFlag;
+        private string _pattern;
+
+        private List<TreeViewItem> _resultItems;
+        public List<TreeViewItem> ResultItems
+        {
+            get
+            {
+                return _resultItems;
+            }
+            set
+            {
+                _resultItems = value;
+                OnPropertyChanged();
+            }
+        }
+
+        private void RecalculateDescription()
+        {
+            if (_pattern.Equals(string.Empty))
+            {
+                _description = "No Pattern given";
+                var results = new List<TreeViewItem>();
+                var rootTreeItem = new TreeViewItem();
+                rootTreeItem.Header = _description;
+                results.Add(rootTreeItem);
+                ResultItems = results;
+                return;
+            }
+            var pattern = new Pattern(_pattern, _ignoreCaseFlag, _ignoreCaseFlag);
+            //_description = pattern.Description;
```

Comment:
because it's not fully cleaned up yet. The "No Pattern given" will be extracted into a .resx properly, too. This is a "relic" of the old way I implemented this